task_three_and_four/src/test/java/com/epam/webdriver/aleksandr_gostev/page/google_cloud/EmailYourEstimatePage.java

```diff
@@ -0,0 +1,50 @@
+package com.epam.webdriver.aleksandr_gostev.page.google_cloud;
+
+import com.epam.webdriver.aleksandr_gostev.page.temp_mail.TempMailPage;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.PageFactory;
+
+import java.awt.datatransfer.UnsupportedFlavorException;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class EmailYourEstimatePage {
+    private final WebDriver driver;
+
+    @FindBy(xpath = "//md-input-container/input[@type='email']")
+    private WebElement emailInput;
+
+    @FindBy(xpath = "//md-dialog-actions/button[@aria-label='Send Email']")
+    private WebElement sendEmailButton;
+
+    public EmailYourEstimatePage(WebDriver driver) {
+        this.driver = driver;
+        PageFactory.initElements(driver, this);
```

Comment:
Please create a common class for all pages that will contain all the common characteristics. For instance, this class could contain a constructor which will have PageFactory.initElements(driver, this);
Also, this class could have methods which will utilise WebDriverWait, i.e. you will not need to use
new WebDriverWait(driver,Duration.ofSeconds(10)).until(ExpectedConditions.elementToBeClickable(xpath));
every time you need to wait for an element, you could just pass your xpath into the appropriate method which will wait for it.
All your pages will inherit this class and, therefore, its functionality. This change will significantly decrease code duplications.